sdk/storage/azure-storage-blob-batch/src/main/java/com/azure/storage/blob/batch/BlobBatch.java

```diff
@@ -0,0 +1,427 @@
+     * Track the status codes expected for the batching operations here as the batch body does not get parsed in
+     * Azure Core where this information is maintained.
+     */
+    private static final int[] EXPECTED_DELETE_STATUS_CODES = {202};
+    private static final int[] EXPECTED_SET_TIER_STATUS_CODES = {200, 202};
+
+    private final ClientLogger logger = new ClientLogger(BlobBatch.class);
+
+    private final BlobAsyncClient blobAsyncClient;
+
+    private final Deque<Mono<? extends Response<?>>> batchOperationQueue;
+    private final List<ByteBuffer> batchRequest;
+    private final Map<Integer, BlobBatchOperationResponse<?>> batchMapping;
+
+    private final AtomicInteger contentId;
+    private final String batchBoundary;
+    private final String contentType;
+
+    private BlobBatchType batchType;
+
+    BlobBatch(String accountUrl, HttpPipeline pipeline) {
+        this.contentId = new AtomicInteger(0);
+        this.batchBoundary = String.format(BATCH_BOUNDARY_TEMPLATE, UUID.randomUUID());
+        this.contentType = String.format(REQUEST_CONTENT_TYPE_TEMPLATE, batchBoundary);
+
+        boolean batchHeadersPolicySet = false;
+        HttpPipelineBuilder batchPipelineBuilder = new HttpPipelineBuilder().httpClient(this::setupBatchOperation);
+        for (int i = 0; i < pipeline.getPolicyCount(); i++) {
+            HttpPipelinePolicy policy = pipeline.getPolicy(i);
+
+            if (policy instanceof SharedKeyCredentialPolicy) {
+                batchHeadersPolicySet = true;
+                // The batch policy needs to be added before the SharedKey policy to run preparation cleanup.
+                batchPipelineBuilder.policies(this::cleanseHeaders, this::setRequestUrl);
+            }
+
+            batchPipelineBuilder.policies(pipeline.getPolicy(i));
+        }
+
+        if (!batchHeadersPolicySet) {
+            batchPipelineBuilder.policies(this::cleanseHeaders, this::setRequestUrl);
+        }
+
+        this.blobAsyncClient = new BlobClientBuilder()
+            .endpoint(accountUrl)
+            .blobName("")
+            .pipeline(batchPipelineBuilder.build())
+            .buildAsyncClient();
+
+        this.batchOperationQueue = new ConcurrentLinkedDeque<>();
+        this.batchRequest = new ArrayList<>();
+        this.batchMapping = new ConcurrentHashMap<>();
+    }
+
+    /**
+     * Adds a delete blob operation to the batch.
+     *
+     * <p><strong>Code sample</strong></p>
+     *
+     * {@codesnippet com.azure.storage.blob.batch.BlobBatch.deleteBlob#String-String}
+     *
+     * @param containerName The container of the blob.
+     * @param blobName The name of the blob.
+     * @return a {@link Response} that will be used to associate this operation to the response when the batch is
+     * submitted.
+     * @throws UnsupportedOperationException If this batch has already added an operation of another type.
+     */
+    public Response<Void> deleteBlob(String containerName, String blobName) {
```

Comment:
Are we not supporting `withResponse` overloads in this?

Reply:
Batching has an interesting implementation where using the `BlobBatch` API instantly returns an empty response until it is submitted via `BlobBatchClient.submitBatch(BlobBatch)`